runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -215,6 +215,7 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
          auto table = (JITClientPersistentCHTable*)comp->getPersistentInfo()->getPersistentCHTable();
          std::string encoded = FlatPersistentClassInfo::serializeHierarchy(table);
 
+         compInfo->getclassesCachedAtServer().clear();
```

Comment:
This require a mutex.
Maybe it's better to put is after the `write` operation so that we don't delay it.